Replaced field_masking_span occurrences with respective ParseField

GrayFox1 opened the pull request:
Closes #63527.

Replaced field_masking_span occurrences with respective ParseField. This change improves maintainability, since some hard coded Strings were replaced by the ParseField that i've defined in the FieldMaskingSpanQueryBuilder Class, besides being consistent with other span queries.

I've ran the tests with "gradlew :server:test --tests org.elasticsearch.index.*" and i would appreciate if someone could give me some feedback.

GrayFox1 committed:
Replaced field_masking_span occurrences with respective ParseField

GrayFox1 committed:
Replaced field_masking_span occurrences with respective

mark-vieira commented:
@elasticmachine retest this please

mark-vieira commented:
@elasticmachine retest this please

mark-vieira commented:
@elasticmachine test this pleasee

romseygeek commented on server/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java:
```diff
@@ -110,7 +110,7 @@ public static FieldMaskingSpanQueryBuilder fromXContent(XContentParser parser) t
                         throw new ParsingException(parser.getTokenLocation(), "[field_masking_span] query must be of type span query");
```
Can we replace the name in these error messages as well?

GrayFox1 replied:
Sure, i'll fix those messages in my next commit

romseygeek commented on server/src/test/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilderTests.java:
```diff
@@ -73,13 +74,13 @@ public void testFromJson() throws IOException {
-            equalTo("field_masking_span [query] as a nested span clause can't have non-default boost value [0.23]"));
+            equalTo(NAME.getPreferredName() + " [query] as a nested span clause can't have non-default boost value [0.23]"));
     }
 
```
Can we also add a test that uses the old deprecated name, and checks for a deprecation warning? There's an `assertWarnings()` helper method that will check for specific deprecation strings.

GrayFox1 replied:
Sure, i'll add a new test in my next commit.

romseygeek reviewed (commented):
Thanks for opening this pull request @GrayFox1! I left a couple of comments but this looks pretty close to me.

romseygeek commented:
@elasticmachine test this please

GrayFox1 committed:
Updates error messages

GrayFox1 committed:
Updates list of expected deprecated queries

GrayFox1 committed:
Adds the test testJsonWithDeprecatedName

GrayFox1 commented:
Thanks for your feedback @romseygeek! I just updated my pull request according to your comments and ran all tests with `gradlew :server:test --tests org.elasticsearch.index.*`

romseygeek commented on .idea/eclipseCodeFormatter.xml:
```diff
@@ -11,4 +11,4 @@
     </option>
   </component>
-</project>
+</project>
```
Nit: can you revert this?

GrayFox1 replied:
Sure, i guess this happened when i was trying to configure the eclipseCodeFormatter plugin, but i'll revert this in my next commit.

romseygeek commented on server/src/test/java/org/elasticsearch/search/SearchModuleTests.java:
```diff
@@ -303,7 +304,7 @@ public void testRegisterRescorer() {
             "dis_max",
             "exists",
-            "field_masking_span",
+            NAME.getPreferredName(),
```
Let's just use "span_field_masking" here to keep in line with the other entries in the list.

romseygeek reviewed (changes requested):
Thanks @GrayFox1, this looks great.  Two more nits and then I think we're done, as long as CI is happy.

romseygeek commented:
@elasticmachine test this please

romseygeek commented:
@GrayFox1 you'll need to merge in the latest version of our master branch to get tests to pass (we've had a version bump internally so all the BWC checks are complaining).

GrayFox1 committed:
Updates list of expected queries

GrayFox1 committed:
Removes unused import and Reverts changes to eclipseCodeFormatter

GrayFox1 committed:
Reverts changes to eclipseCodeFormatter

GrayFox1 committed:
Merge remote-tracking branch 'upstream/master'

GrayFox1 committed:
Fixes expected queries lists alphabetical order

elasticmachine commented:
Pinging @elastic/es-search (Team:Search)

GrayFox1 commented:
Thanks @romseygeek! I just updated my pull request according to your comments, ran the tests with `gradlew :server:test --tests org.elasticsearch.index*` and all of them passed. 

There is only one test failing when i ran `gradlew :server:test --tests org.elasticsearch.search*` in the SearchModuleTests. The failure is on the testRegisteredQueries when it iterates on the expected queries list and can't find "span_field_masking", although i have added this query in the list.

I tried looking into the SearchModule to check if the QuerySpec Registration wasn't using the ParseField, but everything looks ok.

romseygeek commented:
@elasticmachine test this please

romseygeek commented:
Aha, this looks like it's a bug in the test - it's filtering out queries that have *any* deprecated names, not queries that have been entirely deprecated.  I'll open a separate fix for that.

romseygeek commented:
I opened https://github.com/elastic/elasticsearch/pull/74906 to fix the search test issue - you can merge that in here first as well if you like so that we can see how happy CI is otherwise.

romseygeek commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into master

romseygeek commented:
@elasticmachine ok to test

romseygeek commented:
Thanks @GrayFox1!

elasticsearchmachine commented:
## 💔 Backport failed

| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.x  | Commit could not be cherrypicked due to conflicts |

To backport manually run:
`backport --pr 74718`

romseygeek commented:
Manually backported in fc5b80abead122b42a164fd8f80478204cb0674f
